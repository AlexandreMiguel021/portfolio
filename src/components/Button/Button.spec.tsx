import { render, screen } from '@testing-library/react'
import { Button } from '.'
import '@testing-library/jest-dom'
import { App } from 'react-bootstrap-icons'

describe('Button', () => {
	it('should render a Button correctly', () => {
		const { container } = render(
			<Button size={'small'} icon={<App />}>
				sim
			</Button>
		)
		expect(container).toMatchSnapshot()
	})

	it('should render a small Button', () => {
		render(
			<Button size={'small'} icon={<App />}>
				click me
			</Button>
		)
		expect(screen.getByTestId('btn')).toHaveClass('small')
	})

	it('should render a large button', () => {
		render(
			<Button size={'large'} icon={<App />}>
				click me
			</Button>
		)
		expect(screen.getByTestId('btn')).toHaveClass('large')
	})

	it('should render a text content in the button', () => {
		render(
			<Button size={'large'} icon={<App />}>
				click me
			</Button>
		)
		expect(screen.getByText('click me')).toBeInTheDocument()
	})

	it('should render a icon in the button', () => {
		render(
			<Button size={'large'} icon={<App />}>
				click me
			</Button>
		)
		expect(screen.getByTestId('btn').querySelector('svg')).toBeInTheDocument()
	})
})
