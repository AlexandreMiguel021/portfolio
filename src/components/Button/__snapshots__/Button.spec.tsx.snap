// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a Button correctly 1`] = `
<div>
  <button
    class="component small"
    data-testid="btn"
    icon="[object Object]"
  >
    <span
      class="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 2a3 3 0 0 1 3 3v6a3 3 0 0 1-3 3H5a3 3 0 0 1-3-3V5a3 3 0 0 1 3-3h6zM5 1a4 4 0 0 0-4 4v6a4 4 0 0 0 4 4h6a4 4 0 0 0 4-4V5a4 4 0 0 0-4-4H5z"
        />
      </svg>
    </span>
    <span
      class="text"
    >
      sim
    </span>
  </button>
</div>
`;
